src/TGConnection/SocketMessenger/EncryptedSocketMessenger.php

```diff
@@ -363,7 +364,11 @@ private function processServiceMessage(AnonymousMessage $message): void
             $result = $rpcResult->getResult();
 
             if (RpcError::isIt($result)) {
-                $this->analyzeRpcError(new RpcError($result));
+                $this->analyzeRpcError(new RpcError($result), $this->sentMessages[$msgRequestId] ?? null);
+            }
+
+            if (isset($this->sentMessages[$msgRequestId])) {
+                unset($this->sentMessages[$msgRequestId]);
```

Comment:
Wouldn`t it be more smooth to process FLOODs in `getResponseAsync` - where we got request message as well?

Reply:
`getResponseAsync` only gets callbacks, responses are processed here, I don't think we need to move error handling.